src/middleware/notificationOnEditHandler.ts

```diff
@@ -0,0 +1,75 @@
+import autoBind from "auto-bind"
+
+import UserWithSiteSessionData from "@root/classes/UserWithSiteSessionData"
+import { SiteMember, User } from "@root/database/models"
+import CollaboratorsService from "@root/services/identity/CollaboratorsService"
+import NotificationsService from "@root/services/identity/NotificationsService"
+import SitesService from "@root/services/identity/SitesService"
+import ReviewRequestService from "@root/services/review/ReviewRequestService"
+import { RequestHandler } from "@root/types"
+
+export class NotificationOnEditHandler {
+  private readonly reviewRequestService: ReviewRequestService
+
+  private readonly sitesService: SitesService
+
+  private readonly collaboratorsService: CollaboratorsService
+
+  private readonly notificationsService: NotificationsService
+
+  constructor({
+    reviewRequestService,
+    sitesService,
+    collaboratorsService,
+    notificationsService,
+  }: {
+    reviewRequestService: ReviewRequestService
+    sitesService: SitesService
+    collaboratorsService: CollaboratorsService
+    notificationsService: NotificationsService
+  }) {
+    this.reviewRequestService = reviewRequestService
+    this.sitesService = sitesService
+    this.collaboratorsService = collaboratorsService
+    this.notificationsService = notificationsService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  /**
+   * Creates a notification. Requires attachSiteHandler as a precondition
+   */
+  createNotification: RequestHandler<
+    never,
+    unknown,
+    unknown,
+    never,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res, next) => {
+    const { userWithSiteSessionData } = res.locals
+    const { siteName, isomerUserId: userId, email } = userWithSiteSessionData
+    const site = await this.sitesService.getBySiteName(siteName)
+    const users = await this.collaboratorsService.list(siteName, userId)
+    if (!site) throw new Error("Site should always exist")
```

Comment:
out of curiosity, how does this work? if the earlier function did its job, it would have done a `res.status(200).send()`. 

however, this will cause a short circuit to `errorHandler` isn't it and cause a conflicting status code

Reply:
hmm i actually haven't tested this! That said this should be a theoretically unreachable state